jf_agent/git/gitlab_v3_adapter.py

```diff
@@ -0,0 +1,568 @@
+import gitlab3
```

Comment:
It looks like the new adapter and client are starting off as pretty much a copies of the existing adapter and client. That makes sense as a starting point, but you probably want to go one step farther before we ask anybody to run this and look at the [README](https://github.com/doctormo/python-gitlab3) and [API docs ](https://alexvh.github.io/python-gitlab3/gitlab3.GitLab-class.html)for the v3 library. There are some calls in there that look like they have different names than the later library does -- e.g., get_group(project_id) isn't a thing in the v3 library; it looks like it's group(project_id) instead.

Also, it may be that we don't need a separate adapter and client for the v3 library, perhaps we only need a v3 client and can re-use the adapter across both providers? If that would save a lot of code duplication it'd seem worth it, even if you had to make some tweaks to the adapter code here and there to work either client.

Reply:
Removed `gitlab_v3_adapter.py` in this new PR, reviewing the gitlab3 docs from the link above, I think the client files can be identical as well (maybe with a conditional import) depending on how the client-side tests go.